DemoApp/BannerShowingConnectionDelegate.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+final class BannerShowingConnectionDelegate {
+    // MARK: - Private Properties
+    
+    private let navigationController: UINavigationController
+    private let bannerView = BannerView()
+    private let bannerAppearanceDuration: TimeInterval = 0.5
+    
+    // MARK: -
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+        setupViews()
+    }
+}
+
+// MARK: - ChatConnectionControllerDelegate
+
+extension BannerShowingConnectionDelegate: ChatConnectionControllerDelegate {
+    public func connectionController(_ controller: ChatConnectionController, didUpdateConnectionStatus status: ConnectionStatus) {
+        switch status {
+        case .disconnected:
+            showBanner()
+        case .connected:
+            hideBanner()
+        case .initialized,
+             .disconnecting,
+             .connecting:
+            break
+        }
+    }
+}
+
+// MARK: - Private Methods
+
+private extension BannerShowingConnectionDelegate {
+    func setupViews() {
+        attachToTopViewIfNeeded()
+        bannerView.alpha = 0
+        bannerView.update(text: "Connecting...")
+    }
+    
+    func attachToTopViewIfNeeded() {
+        guard let topView = navigationController.topViewController?.view,
+              topView != bannerView.superview else { return }
+        
+        topView.addSubview(bannerView)
+        
+        let guide = topView.safeAreaLayoutGuide
+        NSLayoutConstraint.activate(
+            [
+                bannerView.topAnchor.constraint(equalTo: guide.topAnchor)
+            ]
+        )
+    }
```

Comment:
Rather than messing up with the view, can't we just add the banner to the navbar view?

Reply:
Now it works like this. I was having doubts on this approach when I first considered how to implement it because it is based on an assumption that navbar has clipsToBounds == false and also that it has the highest Z position, but probably it should be fine for the DemoApp. Anyway it's easy to rework if there will be issues